### ISSUE TITLE
Introduce header/data compression for sending events

### ISSUE THREAD

**Issue by JSmith01** [mixpanel/mixpanel-js#439]
The company I work in uses Mixpanel API for quite some time at a decent scale for their products. Some of them send a big amount of data for each event. Additionally, Mixpanel does batching, that leads to requests having huge payloads. The question: is it possible to add some data compression (gzip/etc)?

We've implemented this for our internal tracking tool, so it uses "Content-Encoding: gzip" header and compresses (our case) json payload using [Compression Streams API](https://developer.mozilla.org/en-US/docs/Web/API/CompressionStream). But our tracking backend supports this encoding. In the product we mostly support recent browsers, so this compression API is almost always present.

It is critical for some applications to have an ability to deliver events in a realtime still keeping them small despite the high amount of reported data. As an example, our company provides a WebRTC-based videoconferencing solution, that in some circumstances suffers from additional traffic we're producing by calling the Mixpanel API.

To sum up, please consider adding compression for sending event payloads.
